File: back/src/main/java/com/back/standard/recommend/util/SimilarityCalcer.java
package com.back.standard.recommend.util;

import java.util.List;
import java.util.Map;

public interface SimilarityCalcer {

    record Similar (
            long id,
            double score
    ) { }

    void setVectorA(Vector vector);
    void setVectorB(Vector vector);
    double getSimilarity();

    List<Similar> getSimilarList(Vector target, Map<Long, Vector> matrix);
}

File: back/src/main/java/com/back/standard/recommend/util/SimilarityCalcer.kt
package com.back.standard.recommend.util

interface SimilarityCalcer {

    data class Similar(
        val id: Long,
        val score: Double
    )

    fun setVectorA(vector: Vector)
    fun setVectorB(vector: Vector)
    fun getSimilarity(): Double

    fun getSimilarList(target: Vector, matrix: Map<Long, Vector>): List<Similar>
}